burnman/eos/birch_murnaghan_4th.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import absolute_import
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2015 by the BurnMan team, released under the GNU GPL v2 or later.
+
+
+import scipy.optimize as opt
+from . import equation_of_state as eos
+from ..tools import bracket
+import warnings
+
+def bulk_modulus_fourth(volume, params):
+    """
+    compute the bulk modulus as per the third order
+    birch-murnaghan equation of state.  Returns bulk
+    modulus in the same units as the reference bulk
+    modulus.  Pressure must be in :math:`[Pa]`.
+    """
+    B1 = (params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*(params['Kprime_0']-5.))+(59./9.)
+    B2 = (3.*params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*((3.*params['Kprime_0'])-13.))+(129./9.)
+    B3 = (3.*params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*((3.*params['Kprime_0'])-11.))+(105./9.)
+    B4 = (params['K_0']*params['Kprime_prime_0'])+((params['Kprime_0']-4.)*(params['Kprime_0']-3.))+(35./9.)
+
+    x = params['V_0']/volume
+
+    K = (9./16.)*params['K_0']*( (-5.*B1/3.)*pow(x,-5./3.)+(7.*B2/3.)*pow(x,-7./3.)-3*B3*pow(x,-3.)+(11.*B4/3.)*pow(x,-11./3.))
+    return K
+
+def volume(pressure, params):
+    """
+    Get the birch-murnaghan volume at a reference temperature for a given
+    pressure :math:`[Pa]`. Returns molar volume in :math:`[m^3]`
+    """
+
+    func = lambda x: birch_murnaghan(params['V_0']/x, params) - pressure
+    try:
+        sol = bracket(func, params['V_0'], 1.e-2*params['V_0'])
+    except:
+        raise ValueError('Cannot find a volume, perhaps you are outside of the range of validity for the equation of state?')
+    return opt.brentq(func, sol[0], sol[1])
+
+def volume_fourth_order(pressure,params):
```

Comment:
Have you experimented with how fussy brentq is for this EoS? My intuition is that it would still blow up fairly easily (as with BM3).

I wonder if using the new `bracket` function, as above, would be more robust.